libfreerdp/codec/h264.c

```diff
@@ -59,6 +59,739 @@ static H264_CONTEXT_SUBSYSTEM g_Subsystem_dummy =
+
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "SetGUID(MF_MT_MAJOR_TYPE) failure: 0x%04X", hr);
+			goto error;
+		}
+
+		hr = sys->inputType->lpVtbl->SetGUID(sys->inputType, &MF_MT_SUBTYPE, &MFVideoFormat_H264);
+		
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "SetGUID(MF_MT_SUBTYPE) failure: 0x%04X", hr);
+			goto error;
+		}
+
+		hr = sys->transform->lpVtbl->SetInputType(sys->transform, 0, sys->inputType, 0);
+
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "SetInputType failure: 0x%04X", hr);
+			goto error;
+		}
+
+		hr = mf_find_output_type(sys, &MFVideoFormat_IYUV, &sys->outputType);
+
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "mf_find_output_type failure: 0x%04X", hr);
+			goto error;
+		}
+
+		hr = sys->transform->lpVtbl->SetOutputType(sys->transform, 0, sys->outputType, 0);
+
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "SetOutputType failure: 0x%04X", hr);
+			goto error;
+		}
+
+		hr = mf_create_output_sample(sys);
+
+		if (FAILED(hr))
+		{
+			WLog_ERR(TAG, "mf_create_output_sample failure: 0x%04X", hr);
+			goto error;
+		}
+	}
+	return TRUE;
+
+error:
+	WLog_ERR(TAG, "mf_init failure");
+	mf_uninit(h264);
+	return FALSE;
+}
+
+static H264_CONTEXT_SUBSYSTEM g_Subsystem_MF =
+{
+	"MediaFoundation",
+	mf_init,
+	mf_uninit,
+	mf_decompress,
+	mf_compress
+};
+
+#endif
+
+/**
+ * x264 subsystem
+ */
+
+#ifdef WITH_X264
+
+#define NAL_UNKNOWN	X264_NAL_UNKNOWN
+#define NAL_SLICE	X264_NAL_SLICE
+#define NAL_SLICE_DPA	X264_NAL_SLICE_DPA
+#define NAL_SLICE_DPB	X264_NAL_SLICE_DPB
+#define NAL_SLICE_DPC	X264_NAL_SLICE_DPC
+#define NAL_SLICE_IDR	X264_NAL_SLICE_IDR
+#define NAL_SEI		X264_NAL_SEI
+#define NAL_SPS		X264_NAL_SPS
+#define NAL_PPS		X264_NAL_PPS
+#define NAL_AUD		X264_NAL_AUD
+#define NAL_FILLER	X264_NAL_FILLER
+
+#define NAL_PRIORITY_DISPOSABLE	X264_NAL_PRIORITY_DISPOSABLE
+#define NAL_PRIORITY_LOW	X264_NAL_PRIORITY_LOW
+#define NAL_PRIORITY_HIGH	X264_NAL_PRIORITY_HIGH
+#define NAL_PRIORITY_HIGHEST	X264_NAL_PRIORITY_HIGHEST
+
+#include <stdint.h>
+#include <x264.h>
+
+struct _H264_CONTEXT_X264
+{
+	void* dummy;
+};
+typedef struct _H264_CONTEXT_X264 H264_CONTEXT_X264;
+
+static int x264_decompress(H264_CONTEXT* h264, BYTE* pSrcData, UINT32 SrcSize)
+{
+	//H264_CONTEXT_X264* sys = (H264_CONTEXT_X264*) h264->pSystemData;
+
+	return 1;
+}
+
+static int x264_compress(H264_CONTEXT* h264, BYTE** ppDstData, UINT32* pDstSize)
+{
+	//H264_CONTEXT_X264* sys = (H264_CONTEXT_X264*) h264->pSystemData;
+
+	return 1;
+}
+
+static void x264_uninit(H264_CONTEXT* h264)
+{
+	H264_CONTEXT_X264* sys = (H264_CONTEXT_X264*) h264->pSystemData;
+
+	if (sys)
+	{
+		free(sys);
+		h264->pSystemData = NULL;
+	}
+}
+
+static BOOL x264_init(H264_CONTEXT* h264)
+{
+	H264_CONTEXT_X264* sys;
+
+	sys = (H264_CONTEXT_X264*) calloc(1, sizeof(H264_CONTEXT_X264));
+
+	if (!sys)
+	{
+		goto EXCEPTION;
+	}
+
+	h264->pSystemData = (void*) sys;
+
+	if (h264->Compressor)
+	{
+
+	}
+	else
+	{
+
+	}
```

Comment:
I'm not sure but do those empty if's have any special purpose?
I would suggest if it's something we currently don't handle but need to add in the future using
#if 0 and a comment would be more nice.